### ISSUE TITLE
Alert errors when using graphite queries with "transform" functions

### ISSUE THREAD

**Issue by andrey-makhnutin** [grafana/grafana#37614]
**What happened**:
Alert fired with an error "request handler error: failed to query data: json: cannot unmarshal number into Go struct field TargetResponseDTO.tags of type string"

**What you expected to happen**:
Alert not firing

**How to reproduce it (as minimally and precisely as possible)**:
Create an alert on top of a graphite query with a "scale" function.

**Anything else we need to know?**:
On the alert edit panel, if I press "Test rule" it shows me this:
```
{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "18.294ms",
  "error": "request handler error: failed to query data: json: cannot unmarshal number into Go struct field TargetResponseDTO.tags of type string",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1628194202870,
        "queries": [
          {
            "refId": "A",
            "model": {
              "refCount": 0,
              "refId": "A",
              "target": "scale(minutely.sys-metrics.rucenter.disk-usage, 1024)"
            },
            "datasource": {
              "id": 1,
              "name": "graphite"
            },
            "maxDataPoints": 0,
            "intervalMs": 0
          }
        ],
        "to": 1628196002870
      }
    }
  ]
}
```

If I remove `scale` function from a query, the error disappears.

Query inspector reports this:
```
{
  "request": {
    "method": "POST",
    "url": "api/datasources/proxy/1/render",
    "data": "target=scale(minutely.sys-metrics.rucenter.disk-usage%2C%201024)&target=alias(constantLine(19500000000)%2C%20'limit')&from=-7d&until=now&format=json&maxDataPoints=1412",
    "hideFromInspector": false
  },
  "response": [
    {
      "datapoints": [
        [
          17798070272,
          1627592640000
        ],
        // and a lot more ...
      ],
      "target": "scale(minutely.sys-metrics.rucenter.disk-usage,1024)",
      "tags": {
        "scale": 1024,
        "name": "minutely.sys-metrics.rucenter.disk-usage"
      },
      "title": "scale(minutely.sys-metrics.rucenter.disk-usage,1024)"
    },
    {
      "datapoints": [
        [
          19500000000,
          1627592142000
        ],
        // ...
      ],
      "target": "limit",
      "tags": {
        "name": "19500000000"
      },
      "title": "limit"
    }
  ]
}
```

**Environment**:
- Grafana version: 8.1.0
- Data source type & version: graphite v1.0.x
- OS Grafana is installed on: Ubuntu 18
- User OS & Browser: n/a
- Grafana plugins: Grafana Image Renderer 3.0.1, apart from preinstalled core plugins

**Comment by ifrost** [comment]
Are you sure it's Graphite 1.0.x? Tags were introduced in Graphite 1.1. 

It seems that for Graphite <= 1.1.7 tag values may be returned as numbers, while in 1.1.8 tag values are always strings (fixed by https://github.com/graphite-project/graphite-web/commit/7e0dec19b8062d09a08d6c7ce00bb2f028ba42ec)

**Comment by ifrost** [comment]
So one option could be upgrading to Graphite 1.1.8 or wait for the fix - https://github.com/grafana/grafana/pull/37882

**Comment by andrey-makhnutin** [comment]
Thanks for looking into it, and for fixing the bug! In the end I upgraded my graphite-web to 1.1.8 and it seems to have fixed the issue.

The "graphite v1.0.x" in my first message came from Datasource page of Grafana. It looks like I configured it incorrectly.
It was hard to pinpoint the precise version of graphite-web, because I installed it on my system from source master branch, and in this case the version is always "1.2.0-dev". Judging by modification dates, my previous installation was some time between 1.1.4 and 1.1.5.
